# Do not copy stage0 `<pgm>`s since they are built for host.
# Not copying it will force build system to re-compile them using cross-compiler built in stage0.
--- ghc/hadrian/src/Rules/Program.hs	2023-10-07 08:28:11.000000000 +0530
+++ ghc.mod/hadrian/src/Rules/Program.hs	2024-01-31 18:04:23.053679177 +0530
@@ -100,11 +100,7 @@
   registerPackages =<< contextDependencies ctx
 
   cross <- flag CrossCompiling
-  -- For cross compiler, copy @stage0/bin/<pgm>@ to @stage1/bin/@.
   case (cross, stage) of
-    (True, s) | s > stage0InTree -> do
-        srcDir <- buildRoot <&> (-/- (stageString stage0InTree -/- "bin"))
-        copyFile (srcDir -/- takeFileName bin) bin
     (False, s) | s > stage0InTree && (package `elem` [touchy, unlit]) -> do
         srcDir <- stageLibPath stage0InTree <&> (-/- "bin")
         copyFile (srcDir -/- takeFileName bin) bin
